ql/src/java/org/apache/hadoop/hive/ql/metadata/HiveMetaStoreClientWithLocalCache.java

```diff
@@ -115,687 +46,15 @@ public HiveMetaStoreClientWithLocalCache(Configuration conf, HiveMetaHookLoader
-          rqst.getDb_name(), rqst.getTbl_name(), Warehouse.getPartValuesFromPartName(partitionName),
-          rqst.isGet_col_stats(), rqst.getProcessorCapabilities(), rqst.getProcessorIdentifier(),
-          rqst.getEngine(), rqst.getValidWriteIdList());
-      Partition v = (Partition) cache.get(cacheKey);
-      if (v == null) {
-        partitionsMissing.add(partitionName);
-      } else {
-        if (watermark == null) {
-          LOG.debug(
-              "Query level HMS cache: method=getPartitionsByNamesInternal, dbName={}, tblName={}, partitionName={}",
-              rqst.getDb_name(), rqst.getTbl_name(), partitionName);
-        } else {
-          LOG.debug(
-              "HS2 level HMS cache: method=getPartitionsByNamesInternal, dbName={}, tblName={}, partitionName={}",
-              rqst.getDb_name(), rqst.getTbl_name(), partitionName);
-        }
-        partitions.add(v);
-      }
-    }
-    return Pair.of(partitions, partitionsMissing);
-  }
-
-  protected final List<Partition> loadPartitionsByNamesCache(CacheI cache,
-      GetPartitionsByNamesResult r, GetPartitionsByNamesRequest rqst, TableWatermark watermark) {
-    List<Partition> newPartitions = new ArrayList<>();
-    for (Partition partition : r.getPartitions()) {
-      CacheKey cacheKey = new CacheKey(KeyType.PARTITIONS_BY_NAMES, watermark,
-          rqst.getDb_name(), rqst.getTbl_name(), partition.getValues(),
-          rqst.isGet_col_stats(), rqst.getProcessorCapabilities(), rqst.getProcessorIdentifier(),
-          rqst.getEngine(), rqst.getValidWriteIdList());
-      cache.put(cacheKey, partition);
-      newPartitions.add(partition);
-    }
-    return newPartitions;
+    super(createUnderlyingClient(conf, hookLoader, allowEmbedded));
   }
 
-  protected final GetPartitionsByNamesResult computePartitionsByNamesFinal(GetPartitionsByNamesRequest rqst,
-      List<Partition> partitions, List<Partition> newPartitions) throws MetaException {
-    List<Partition> result = new ArrayList<>();
-    int i = 0, j = 0;
-    for (String partitionName : rqst.getNames()) {
-      if (i >= partitions.size() || j >= newPartitions.size()) {
-        break;
-      }
-      List<String> pv = Warehouse.getPartValuesFromPartName(partitionName);
-      if (partitions.get(i).getValues().equals(pv)) {
-        result.add(partitions.get(i));
-        i++;
-      } else if (newPartitions.get(j).getValues().equals(pv)) {
-        result.add(newPartitions.get(j));
-        j++;
-      }
-    }
-    while (i < partitions.size()) {
-      result.add(partitions.get(i));
-      i++;
-    }
-    while (j < newPartitions.size()) {
-      result.add(newPartitions.get(j));
-      j++;
-    }
-    return new GetPartitionsByNamesResult(result);
+  private static IMetaStoreClient createUnderlyingClient(Configuration conf, HiveMetaHookLoader hookLoader,
+      Boolean allowEmbedded) throws MetaException {
+    IMetaStoreClient thriftClient = new ThriftHiveMetaStoreClient(conf, allowEmbedded);
```

Comment:
If we have ThriftHiveMetaStoreClient here it means we can not reuse the functionality of this class with other than ThriftHiveMetaStoreClient e.g. RESTClient for Iceberg REST catalog can not reuse the functionality of ql/HiveMetaStoreClientWithLocalCache. This should be an input as delegate I guess.